src/background/background-backup-wallet-salt-handler.ts

```diff
@@ -0,0 +1,88 @@
+import { logger } from '@shared/logger';
+import { gaiaUrl } from '@shared/constants';
+import { createWalletGaiaConfig, generateWallet } from '@stacks/wallet-sdk';
+import { GaiaHubConfig, uploadToGaiaHub } from '@stacks/storage';
+import { decryptContent, encryptContent, getPublicKeyFromPrivate } from '@stacks/encryption';
+import { fetchPrivate } from '@stacks/common';
+
+interface WalletDataMigrationConfig {
+  version: string;
+  oldWalletSalt: string;
+  meta?: {
+    [key: string]: any;
+  };
+}
+
+async function fetchWalletMigrationData({
+  configPrivateKey,
+  gaiaHubConfig,
+}: {
+  configPrivateKey: string;
+  gaiaHubConfig: GaiaHubConfig;
+}) {
+  try {
+    const response = await fetchPrivate(
+      `${gaiaHubConfig.url_prefix}${gaiaHubConfig.address}/wallet-migration-data.json`
+    );
+    if (!response.ok) return null;
+    const encrypted = await response.text();
+    const configJSON = (await decryptContent(encrypted, {
+      privateKey: configPrivateKey,
+    })) as string;
+    const config: WalletDataMigrationConfig = JSON.parse(configJSON);
+    return config;
+  } catch (error) {
+    logger.error(error);
+    return null;
+  }
+}
+
+const walletMigrationFilename = 'wallet-migration-data.json';
+const walletMigrationVersion = '1.0';
+
+async function storeToGaia(configPrivateKey: string, gaiaHubConfig: GaiaHubConfig, salt: string) {
```

Comment:
should we use different naming for the salt arg here?

Reply:
I think salt is fine here, this is what we want to store. I don't want to add the version also here in the name and neither use 'old'. If you have any suggestion let me know otherwise I will keep `salt`.

Reply:
I posted my last comment in the wrong place, sorry, meant to do here ...Again, you could use this here walletSaltThatGeneratesIncorrectAppKey?

Reply:
I will also use this `walletSaltThatGeneratesIncorrectAppKey` here then.